test/diagnostics/warnings_as_errors_rules.swift

```diff
@@ -0,0 +1,65 @@
+// RUN: not %target-swift-frontend -typecheck -diagnostic-style llvm -warnings-as-errors %s 2>&1 | %FileCheck %s --check-prefix=CHECK-WAE-ALL
```

Comment:
Why is this forcing the LLVM diagnostic style?

Reply:
With the swift style the compiler produces a message like this:
```
$ ./swift-frontend -typecheck -diagnostic-style swift -warnings-as-errors  ../../../../swift/test/diagnostics/warnings_as_errors_rules.swift
../../../../swift/test/diagnostics/warnings_as_errors_rules.swift:42:1: error: 'foo()' is deprecated
40 | // CHECK-WAE-GROUP-NWAE-DIAGID: warning: 'foo()' is deprecated [group:availability_deprecated]
41 | // CHECK-WAE-GROUP-NWAE-DIAGID-NOT: error: 'foo()' is deprecated [group:availability_deprecated]
42 | foo()
   | `- error: 'foo()' is deprecated
43 |
44 |
```
As you can see there are two lines more above and below the line 42. FileCheck sees it in its input and when it performs a `-NOT` check these comments actually produce a match (which fails the testcase). 
Ideally I'd prefer if FileCheck ignored lines containing `.*CHECK-.*`, but I don't see a flag or option for this.

Reply:
Ah, I understand. In our call through to the swift-syntax diagnostic mechanism, we're hard-coding the context size to 2 lines:

```
  swift_ASTGen_renderQueuedDiagnostics(queuedDiagnostics, /*contextSize=*/2,
                                       forceColors ? 1 : 0,
                                       &bridgedRenderedString);
```

... we should have an option here to allow setting the context size (e.g., to 0), which would make FileCheck tests like this more feasible.

Reply:
To be clear, I don't mean "you have to add this other command-line argument yourself." I'm fine with your PR using `-diagnostic-style llvm` until we get the other command-line argument. I suspect that's the reason for some of the other existing uses of `-diagnostic-style llvm`